Netkan/Sources/Gitlab/GitlabApi.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+using CKAN.NetKAN.Services;
+
+namespace CKAN.NetKAN.Sources.Gitlab
+{
+    /// <summary>
+    /// Provides convenient access to the GitLab API
+    /// https://docs.gitlab.com/ee/api/
+    /// </summary>
+    internal sealed class GitlabApi : IGitlabApi
+    {
+        /// <summary>
+        /// Initialize the API object
+        /// </summary>
+        /// <param name="http">HTTP service for getting URLs</param>
+        /// <param name="token">GitLab API token</param>
+        public GitlabApi(IHttpService http, string token = null)
+        {
+            this.http  = http;
+            this.token = token;
```

Comment:
Do we need to to setup a gitlab account for the indexer? Or will we get by until we reach their API limit?

Reply:
I think we will want to create a GitLab token for this just for completeness/robustness, but I tested it without a token and it still worked. If I'm reading this table correctly, a token boosts the rate limiting from 500 to 2000 requests per minute:

https://docs.gitlab.com/ee/user/gitlab_com/index.html#gitlabcom-specific-rate-limits